meteor/server/api/deviceTriggers/StudioDeviceTriggerManager.ts

```diff
@@ -0,0 +1,227 @@
+import { ShowStyleBaseId, StudioId } from '@sofie-automation/corelib/dist/dataModel/Ids'
+import { Complete, getRandomString, literal } from '@sofie-automation/corelib/dist/lib'
+import { applyAndValidateOverrides } from '@sofie-automation/corelib/dist/settings/objectWithOverrides'
+import { ReadonlyObjectDeep } from 'type-fest/source/readonly-deep'
+import {
+	createAction,
+	isPreviewableAction,
+	ReactivePlaylistActionContext,
+} from '../../../lib/api/triggers/actionFactory'
+import { DeviceTriggerMountedActionId, PreviewWrappedAdLibId } from '../../../lib/api/triggers/MountedTriggers'
+import { isDeviceTrigger } from '../../../lib/api/triggers/triggerTypeSelectors'
+import { DBTriggeredActions, UITriggeredActionsObj } from '../../../lib/collections/TriggeredActions'
+import { DummyReactiveVar, protectString } from '../../../lib/lib'
+import { logger } from '../../logging'
+import { GlobalTriggerManager } from './GlobalTriggerManager'
+import { DeviceTriggerMountedActionAdlibsPreview, DeviceTriggerMountedActions } from './observer'
+import { ContentCache } from './reactiveContentCache'
+
+export class StudioDeviceTriggerManager {
+	#showStyleBaseId: ShowStyleBaseId | null = null
+
+	constructor(public studioId: StudioId) {}
+
+	public get showStyleBaseId(): ShowStyleBaseId | null {
+		return this.#showStyleBaseId
+	}
+
+	public set showStyleBaseId(value: ShowStyleBaseId | null) {
+		this.#showStyleBaseId = value
+	}
+
+	updateTriggers(cache: ContentCache | null): void {
```

Comment:
minor note: 
this method could perhaps be split into two, for clarity?
* `updateTriggers`
* `clearTriggers` (using the code path where the cache is `null`)
I noticed that it is only ever called with a cache OR with `null`